### ISSUE TITLE
Unable to use applyMiddleware (from graphql-middleware) after upgrading to 2.2.0

### ISSUE THREAD

**Issue by hinsxd** [apollographql/apollo-server#1934]
With a minimum graphql server with express, apollo-server 2.2.0 throws `Error: Unable to generate server introspection document.` while trying to apply middleware (`graphql-shield`) on the server with `graphql-middleware` 

```
const { ApolloServer } = require('apollo-server-express');
const { makeExecutableSchema } = require('graphql-tools');
const { applyMiddleware } = require('graphql-middleware');
const { shield } = require('graphql-shield');

const typeDefs = `
	type Query {
		me: String!
	}

	type Mutation {
		test: String!
	}
`;
const resolvers = {
	Query: {
		me: () => 'me'
	},
	Mutation: {
		test: () => 'hi'
	}
};

const permissions = shield({
	Query: {
		me: true
	},
	Mutation: {
		test: true
	}
});

const schema = applyMiddleware(
	makeExecutableSchema({ typeDefs, resolvers }),
	permissions
);
const server = new ApolloServer({
	mocks: false,
	schema,
	context: ({ req }) => ({
		...req
	})
});
const app = express();
server.applyMiddleware({ app, path: '/graphql' });
app.listen({ port: 4000 }, () => {
	console.log(`🚀 Server ready at http://localhost:4000`);
});

```
which gives the following error:
```
/Users/hinsxd/projects/askmrfox/test/node_modules/apollo-server-core/dist/utils/schemaHash.js:16
        throw new Error('Unable to generate server introspection document.');
        ^

Error: Unable to generate server introspection document.
    at Object.generateSchemaHash (/Users/hinsxd/projects/askmrfox/test/node_modules/apollo-server-core/dist/utils/schemaHash.js:16:15)
    at new ApolloServerBase (/Users/hinsxd/projects/askmrfox/test/node_modules/apollo-server-core/dist/ApolloServer.js:157:40)
    at new ApolloServer (/Users/hinsxd/projects/askmrfox/test/node_modules/apollo-server-express/dist/ApolloServer.js:44:1)
    at Object.<anonymous> (/Users/hinsxd/projects/askmrfox/test/src/index.js:38:16)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by Infonautica** [comment]
Same problem. Pinned version "apollo-server": "2.0.5" works great

**Comment by wmertens** [comment]
Same problem, but I don't use middlewares.

The problem seems to be that the result of `execute(schema, documentAST)` is a Promise, but it is not awaited.

**Comment by michieldewilde** [comment]
Running into this as well

**Comment by hinsxd** [comment]
It is even more weird that, even if I only declare the middleware but not using it in initializing the server, it still shows the error. Any clue?

**Comment by wmertens** [comment]
@juank11memphis 	@hinsxd 	see #1935 	. You can go back to v2.0.5 in the meantime.

**Comment by hexlyci** [comment]
Confirmed I'm hitting this too as a result from a promise being rerturned in `generateSchemaHash`. Thanks for the pinning workaround – spent a good 10 hours digging and learning (which isn't a wash)

**Comment by dwelch2344** [comment]
Confirmed I'm hitting this too as a result from a promise being rerturned in `generateSchemaHash`. Thanks for the pinning workaround – spent a good 10 hours digging and learning (which isn't a wash)

**Comment by ellipticaldoor** [comment]
Yep, I have the same error

**Comment by juank11memphis** [comment]
I tried using version 2.2.1 thinking the problem was going to be fixed, but I am still getting the same error. This should not be a version with breaking changes, I would expect just to update the version and everything to keep working as before...

**Comment by juank11memphis** [comment]
Version 2.2.2 still not working, it just shows kind of a better error but not good enough to understand what do we need to do to fix this, and still this is not supposed to be a version with breaking changes, and if it is, a little guide would be more than appreciated.
Thanks.

**Comment by ellipticaldoor** [comment]
I think the error comes from here [https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-core/src/utils/schemaHash.ts#L23](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-core/src/utils/schemaHash.ts#L23)

Though, Im not sure what I'm doing wrong, is this type checking my typescript code on execution?

**Comment by abernix** [comment]
Closing as a duplicate of #1935.  I've responded over there as to the problem, but the short explanation is that the execution of an introspection query, which is not supposed to run asynchronously.  The situation occurring here is that resolvers are being wrapped in a way which doesn't maintain their original execution dynamics, thus violating this assumption.  See related: https://github.com/graphql/graphql-js/pull/1120.
